src/main/java/com/nike/cerberus/auth/connector/okta/OktaAuthConnector.java

```diff
@@ -44,44 +53,73 @@
 
     private final OktaClientResponseUtils oktaClientResponseUtils;
 
-    private static final ImmutableSet UNSUPPORTED_OKTA_MFA_TYPES = ImmutableSet.of("push", "call", "sms");
+    private final AuthenticationClient client;
 
     @Inject
     public OktaAuthConnector(final OktaApiClientHelper oktaApiClientHelper,
-                             final OktaClientResponseUtils oktaClientResponseUtils) {
+                             final OktaClientResponseUtils oktaClientResponseUtils,
+                             @Named("auth.connector.okta.base_url") String oktaUrl,
+                             @Named("auth.connector.okta.api_key") String oktaApiKey) {
+
+        System.setProperty("okta.client.token", oktaApiKey);
+
+        client = AuthenticationClients.builder()
+                .setOrgUrl(oktaUrl)
+                .build();
+
+        this.oktaApiClientHelper = oktaApiClientHelper;
+        this.oktaClientResponseUtils = oktaClientResponseUtils;
+    }
+
+//    For testing
+    public OktaAuthConnector(final OktaApiClientHelper oktaApiClientHelper,
```

Comment:
lets move 

```java
client = AuthenticationClients.builder()
                .setOrgUrl(oktaUrl)
                .build();
```

to the CMS Guice module and make it constructor injected and remove this second constructor.

See https://github.com/Nike-Inc/cerberus-management-service/blob/master/src/main/java/com/nike/cerberus/server/config/guice/CmsGuiceModule.java#L240-L244 as an example.

Reply:
This will allow you to mock it just like any other DI class is, without the need to create additional test only constructors.

Reply:
- [ ] remove second constructor and create the client in the CMS Guice module and inject it into this class as a constructor argument.